addons/im_livechat/static/src/embed/common/chat_window_patch.js

```diff
@@ -1,18 +1,41 @@
 import { FeedbackPanel } from "@im_livechat/embed/common/feedback_panel/feedback_panel";
+import { CloseConfirmation } from "@im_livechat/embed/common/close_confirmation";
 
 import { ChatWindow } from "@mail/core/common/chat_window";
 
-import { useState } from "@odoo/owl";
+import { useState, toRaw } from "@odoo/owl";
 
 import { useService } from "@web/core/utils/hooks";
 import { patch } from "@web/core/utils/patch";
 
-Object.assign(ChatWindow.components, { FeedbackPanel });
+Object.assign(ChatWindow.components, { FeedbackPanel, CloseConfirmation });
 
 patch(ChatWindow.prototype, {
     setup() {
         super.setup(...arguments);
         this.livechatService = useService("im_livechat.livechat");
         this.chatbotService = useState(useService("im_livechat.chatbot"));
+        this.livechatState = useState({
+            showCloseConfirmation: false,
+        });
+    },
+
+    async close() {
+        const chatWindow = toRaw(this.props.chatWindow);
+        if (!this.livechatState.showCloseConfirmation) {
+            if (chatWindow.folded) {
+                this.props.chatWindow.open({ notifyState: this.thread?.state !== "open" });
+            }
+            this.livechatState.showCloseConfirmation = true;
+        } else {
+            await super.close();
+        }
+        if (this.closeFeedback) {
+            await super.close();
+        }
```

Comment:
So with previous comment, code can simply be:
```js
async close() {
        if (
            this.livechatService.state === SESSION_STATE.PERSISTED &&
            !this.livechatState.showCloseConfirmation
        ) {
            this.livechatState.showCloseConfirmation = true;
        } else {
            await super.close();
        }
    },
```

Calling it twice will leave the conversation.

To prevent clicking on "X" twice, we can disable all chat window actions. This can be done by adding a `state.actionsDisabled` in chat window, put conditional in main `mail` template, and set it similarly to `livechatState.showCloseConfirmation`